Add no-kubernetes-deployment guide (#1618)
llm-d/llm-d: docs/well-lit-paths/no-kubernetes-deployment.md

# No-Kubernetes Deployment

> llm-d's reference deployment runs on Kubernetes — workers live in `Deployments`, the EPP discovers them through an `InferencePool`, and the platform handles networking and lifecycle. Many environments don't have a Kubernetes control plane, though: HPC schedulers like Slurm or LSF launch workers dynamically, Ray-based stacks run workers as actors, bare-metal inference farms operate without K8s, and a single workstation with a couple of GPUs is often enough for development.

Reviewer: ```suggestion
llm-d's reference deployment runs on Kubernetes — workers are managed by Kubernetes `Deployments`, the EPP discovers them through an `InferencePool`, and the platform handles networking and lifecycle. Many environments don't have a Kubernetes control plane, though: HPC schedulers like Slurm or LSF launch workers dynamically, Ray-based stacks run workers as actors, bare-metal inference farms operate without K8s, and a single workstation with a couple of GPUs is often enough for development.
```

Author: Fixed f50c2c5

> The No-Kubernetes path runs the same routing stack — the EPP, Envoy, and one or more vLLM workers — directly as host processes or containers. The EPP gets its endpoint inventory from a YAML file on disk via the [file-discovery plugin][filediscovery-plugin] instead of watching an `InferencePool` over the Kubernetes API; everything else (EPP plugin set, scoring, Envoy ext_proc, vLLM arguments) is unchanged.

Reviewer: ```suggestion
The No-Kubernetes path runs the same routing stack — the llm-d EPP, Envoy, and one or more model servers — directly as host processes or containers. The EPP gets its endpoint inventory from a YAML file on disk via the [file-discovery plugin][filediscovery-plugin] instead of watching an `InferencePool` over the Kubernetes API; everything else (EPP plugin set, scoring, Envoy ext_proc, vLLM arguments) is unchanged.
```

Author: Fixed f50c2c5

> Background: ["No Kubernetes? No Problem: llm-d Now Runs Anywhere"][blog].

Reviewer: please move this to further reading

Author: Fixed f50c2c5

> [!IMPORTANT]

> > Without Kubernetes, several pieces of the optimized-baseline stack are

Reviewer: ```suggestion
> Without Kubernetes, some pieces of the llm-d stack are
```

Author: Fixed f50c2c5

> out of scope: `InferenceObjective`-driven FlowControl, the
> `InferenceModelRewrite` model-name rewriter, and `PodMonitor`-based
> Prometheus discovery. Scoring, prefix-cache affinity, saturation-based
> admission, and Prometheus metrics on `--metrics-port` all work; see
> the [parity caveats][blog-parity] for the full list.

## Deploy

See the [no-Kubernetes deployment guide](../../guides/no-kubernetes-deployment) for manifests and step-by-step deployment.

## Architecture

<p align="center">
  <picture>
    <source media="(prefers-color-scheme: dark)">
    <img src="../assets/no-kubernetes-deployment.svg" alt="No-Kubernetes Deployment">
  </picture>

> </p>

Reviewer: pls use the same theme as the drawings in https://drive.google.com/corp/drive/folders/1-CyochqqoMccMrZ7BFYPZwg1XRHdS4DY; please also share it with Rob or myself so we add it to the drive for future maintenance

Author: > ls use the same theme as the drawings 

I changed and added to the drive (didn't have a permission issue) .  28f7dca
